rosserial_arduino/src/ros_lib/ArduinoTcpHardware.h

```diff
@@ -30,49 +32,63 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef ESP8266HARDWARE_H
-#define ESP8266HARDWARE_H
+#ifndef ROS_ARDUINO_TCP_HARDWARE_H_
+#define ROS_ARDUINO_TCP_HARDWARE_H_
 
-#include <ESP8266WiFi.h>
+#include <Arduino.h>
+#if defined(ESP8266)
+  #include <ESP8266WiFi.h>
+#else
+  #include <SPI.h>
+  #include <Ethernet.h>
+#endif
 
-class Esp8266Hardware {
+class ArduinoHardware {
   public:
-    Esp8266Hardware()
+    ArduinoHardware()
     {
     }
-    
-    void setConnection(IPAddress &server, int port) {
-      this->server = server;
-      this->serverPort = port;
+
+    void setConnection(IPAddress &server, int port = 11411) {
+      server_ = server;
+      serverPort_ = port;
     }
-    
+
     IPAddress getLocalIP() {
-      return tcp.localIP();
+#if defined(ESP8266)
+      return tcp_.localIP();
+#else
+      return Ethernet.localIP();
+#endif
     }
 
-    void init() { 
-      this->tcp.connect(this->server, this->serverPort);
+    void init(){
+      tcp_.connect(server_, serverPort_);
     }
 
-    int read() {
-      if (this->tcp.connected()) {
-        return tcp.read();
-      } else {
-        this->tcp.connect(this->server, this->serverPort);
+    int read(){
+      if(tcp_.connected()){
+          return tcp_.read();
+      }else{
+        tcp_.connect(server_, serverPort_);
       }
```

Comment:
I know you're just following what was here before, but given that the file is basically being rewritten, it would be great if you could follow ROS brace style: http://wiki.ros.org/CppStyleGuide#Formatting

The astyle tool can quickly update the file: http://wiki.ros.org/roslint#Tips

Reply:
Ok, no problem, I will update the style.